docs/docker-images/all-in-one.md

```diff
@@ -0,0 +1,76 @@
+This image contains `earthly`, `buildkit`, and some extra configuration to enable the two to work together. All that's missing is your source code! This image is mainly intended for use in containerized CI scenarios, or where maintaining a persistent installation of `earthly` isn't possible.
+
+## Quickstart
+
+Want to just get started? Here are a couple sample `docker run` commands that cover the most common use-cases:
+
+### Simple Usage
+
+```bash
+$ docker run --privileged -t -e NO_DOCKER=1 -v $(pwd):/workspace earthly/earthly:latest +for-linux
+```
+
+Heres a quick breakdown:
+
+- `--privileged` is required when you are using the internal `buildkit`. This is because `buildkit` currently requires is when used with `earthly`.
+- `-t` tells Docker to emulate a TTY. This makes the `earthly` log output colorized.
+- `-e NO_DOCKER=1` skips the check for a functional Docker daemon. If you are not exporting any images, you do not need Docker.
+- `-v $(pwd):/workspace` mounts the source code into the conventional location within the docker container. Earthly is executed from this directory when starting the container. Any artifacts saved within this folder remain on your local machine.
+- `+for-linux` is the target to be invoked. All arguments specified after the image tag will be passed to `earthly`.
+
+### More Complicated Usage
+
+```bash
+$ docker run -t --privileged -v $(pwd):/workspace:rw --network=host -v ~/.earthly/config.yml:/etc/.earthly/config.yml -e DOCKER_HOST="tcp://0.0.0.0:2375" earthly/earthly:corey_entrypoint --ci -P +for-linux
```

Comment:
It's more common to use `/var/run/docker.sock` for Docker. Would that be a better example?

Reply:
Yes; but why would you when you are already running privileged? The earlier example works locally; this was intended to show you could use an external `DOCKER_HOST`, as you might in a CI.

Reply:
I tweaked the address and wording a bit to make this less host-centric, at the expense of the command not working out of the box for a user.

Reply:
> Yes; but why would you when you are already running privileged?

One reason is that Docker automatically creates `/var/run/docker.sock`, but it doesn't automatically listen on TCP.

I think both are good examples. Perhaps the table mentioning DOCKER_HOST could mention both examples.

Reply:
I guess what I was trying to get at is that if you are already privileged, isn't best practice to run your own daemon? I'll add more details in the table if that works for you rather than adding another quickstart with `-v /var/run/docker.sock:/var/run/docker.sock`?

Reply:
Yeah that makes sense